pages/2_Workflow.py

```diff
@@ -1,29 +1,126 @@
 import streamlit as st
+import threading
 
 from src.common import *
 from src.workflow import *
+from src.run_subprocess import *
+from src.captcha_ import *
 
 # Page name "workflow" will show mzML file selector in sidebar
 params = page_setup(page="workflow")
+
+#if local no need captcha
+if st.session_state.location == "local":
+    params["controllo"] = True
+    st.session_state["controllo"] = True
+
+#if controllo is false means not captcha applied
+if 'controllo' not in st.session_state or params["controllo"] == False:
+    #apply captcha
+    captcha_control()
+        
+### main content of page
 st.title("Workflow")
 
+tabs = ["Table example", "Run Subprocess example"]
+
+tabs = st.tabs(tabs)
+
 # Define two widgets with values from paramter file
 # To save them as parameters use the same key as in the json file
 
-# We access the x-dimension via local variable
-xdimension = st.number_input(
-    label="x dimension", min_value=1, max_value=20, value=params["example-x-dimension"], step=1, key="example-x-dimension")
+with tabs[0]:
+    # We access the x-dimension via local variable
+    xdimension = st.number_input(
+        label="x dimension", min_value=1, max_value=20, value=params["example-x-dimension"], step=1, key="example-x-dimension")
+
+    st.number_input(label="y dimension", min_value=1, max_value=20,
+                    value=params["example-y-dimension"], step=1, key="example-y-dimension")
+
+    # Get a dataframe with x and y dimensions via time consuming (sleep) cached function
+    # If the input has been given before, the function does not run again
+    # Input x from local variable, input y from session state via key
+    df = generate_random_table(xdimension, st.session_state["example-y-dimension"])
+
+    # Display dataframe via custom show_table function, which will render a download button as well
+    show_table(df, download_name="random-table")    
+
+with tabs[1]:
+    ###### Here just example for run subprocess 
+    mzML_dir: Path = Path(st.session_state.workspace, "mzML-files")
```

Comment:
why not put on streamlit app some text that describes what you want to show? that way it is properly documented in the source code and shown in the app.
e.g., 
I would probably write something like.

This example demonstrates how to run an external process (here the Linux command "grep") as a subprocess
and display the process output. Also works with longer running process like e.g., calling an OpenMS TOPP tool